src/components/controller.cpp

```diff
@@ -386,36 +390,195 @@ void controller::process_eventqueue() {
   }
 }
 
+/**
+ * Tries to match the given command to a legacy action string and sends the
+ * appropriate new action (and data) to the right module if possible.
+ *
+ * \returns true iff the given command matches a legacy action string and was
+ *          successfully forwarded to a module
+ */
+bool controller::try_forward_legacy_action(const string cmd) {
```

Comment:
Should take a const ref I think

Reply:
Yes that works here, I'm not directly modifying cmd